docs/source/basics/custom_components.rst

```diff
@@ -0,0 +1,171 @@
+.. _custom_components_label:
+
+=============================
+Custom Components and Recipes
+=============================
+
+torchtune lets you launch fine-tuning jobs directly from the command-line using both library and custom datasets,
+models, recipes, and configs. This is done with the ``tune run`` command (see :ref:`cli_label`), which can also be
+used from your project folder.
+
+Setting up your torchtune project
+---------------------------------
+First, ensure that you have torchtune installed, see :ref:`install_label`. This will also install the ``tune`` command
+to your environment.
+
+You can launch with ``tune run`` from any directory. Let's create a new project directory and ensure we can launch a built-in
+library recipe with a library config from that folder.
+
+.. code-block:: bash
+
+    mkdir ~/my_project
+    cd ~/my_project
+    # This downloads the Llama 3.2 1B Instruct model
+    tune download meta-llama/Llama-3.2-1B-Instruct --output-dir /tmp/Llama-3.2-1B-Instruct --ignore-patterns "original/consolidated.00.pth"
+    # This launches a lora finetuning run with the default single device config
+    tune run lora_finetune_single_device --config llama3_2/1B_lora_single_device
+
+Launching custom configs
+------------------------
+Often, you'll want to start with one of our default configs for a particular model and adjust a few training hyperparameters.
+You can use the ``tune cp`` command to create a copy of a default config in your project directory so you can make modifications.
+
+.. code-block:: bash
+
+    # Show all the default model configs for each recipe
+    tune ls
```

Comment:
tune ls output be like 
![image](https://github.com/user-attachments/assets/1dc97aa4-3d24-408e-b4f0-8576ad3a59d8)

Reply:
more like
![image](https://github.com/user-attachments/assets/fd47e382-586d-4ddc-a8d5-971739dc5b25)